### PR TITLE
feat: add `array/base/cuevery-by` package

### LINKED ISSUE #2324: [RFC]: add `@stdlib/array/base/cuevery-by`
### Description

This RFC proposes adding the package `@stdlib/array/base/cuevery-by`, which cumulatively tests whether every array element in a provided array passes a test implemented by a predicate function. The function should return a new generic array. The package should also provide an `#assign` API for setting output values in a provided output array.

```javascript
function isPositive( value ) {
	return ( value > 0 );
}

var x = [ 1, 1, 1, 0, 1 ];

var y1 = cueveryBy( x, isPositive );
// returns [ true, true, true, false, false ]

var y2 = [ false, null, false, null, false, null, false, null, false, null ];
var out = cueveryBy.assign( x, y2, 2, 0, isPositive );
// returns [ true, null, true, null, true, null, false, null, false, null ]

var bool = ( out === y2 );
// returns true
```

where the `assign` API supports an offset and stride (see, e.g., `@stdlib/array/base/take`).

Both APIs should support accessor arrays (see, e.g., `@stdlib/array/base/take`).

### Related Issues

No.

### Questions

No.

### Other

- See also `@stdlib/array/base/every-by`

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Kaif987**
This commit adds `base/cuevery-by` package

Resolves #2324

## Description

> What is the purpose of this pull request?
The purpose of this pull request is to add the `@stdlib/base/cuevery-by` package which cumulatively tests whether every array element in a provided array passes a test implemented by a predicate function. 

## Related Issues

> Does this pull request have any related issues?
#2324 

This pull request:

-   resolves #2324 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by Kaif987** [comment]
Hey @kgryte can you review this? let me know if any further changes are needed,

File: lib/node_modules/@stdlib/array/base/cuevery-by/README.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-->
+
+# cuevery-by
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/array/base/cuevery-by/README.md]:
```suggestion
# cueveryBy
```

File: lib/node_modules/@stdlib/array/base/cuevery-by/README.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+## Usage
+
+```javascript
+var cueveryBy = require( '@stdlib/array/base/cuevery-by' );
+```
+
+#### cueveryBy( x, predicate\[, thisArg ] )
+
+Cumulatively test whether every array element in a provided array passes a test implemented by a predicate function.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/array/base/cuevery-by/README.md]:
```suggestion
Cumulatively tests whether every array element in a provided array passes a test implemented by a predicate function.
```

File: lib/node_modules/@stdlib/array/base/cuevery-by/lib/index.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+// MODULES //
+
+var setReadOnly = require( '@stdlib/utils/define-nonenumerable-read-only-property' );
+var main = require( './main.js' );
+var assign = require( './assign.js' );
+
+
+// MAIN //
+
+setReadOnly(main, 'assign', assign);
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/array/base/cuevery-by/lib/index.js]:
```suggestion
setReadOnly( main, 'assign', assign );
```

File: lib/node_modules/@stdlib/array/base/cuevery-by/lib/index.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+/**
+* Cumulatively tests whether every array element in a provided array passes a test implemented by a predicate function.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/array/base/cuevery-by/lib/index.js]:
```suggestion
* Cumulatively test whether every array element in a provided array passes a test implemented by a predicate function.
```

File: lib/node_modules/@stdlib/array/base/cuevery-by/package.json [hunk 5]
```diff
@@ -0,0 +1,62 @@
+{
+  "name": "@stdlib/array/base/cuevery-by",
+  "version": "0.0.0",
+  "description": "cumulatively tests whether every array element in a provided array passes a test implemented by a predicate function",
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/array/base/cuevery-by/package.json]:
```suggestion
  "description": "Cumulatively test whether every array element in a provided array passes a test implemented by a predicate function.",
```

File: lib/node_modules/@stdlib/array/base/cuevery-by/README.md [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+
+var x = [ 1, 1, 0, 0, 0 ];
+
+var y = cueveryBy( x, fcn );
+// returns [ true, true, false, false, false ]
+```
+
+#### cueveryBy.assign( x, out, stride, offset, predicate\[, thisArg ] )
+
+Cumulatively test whether every array element in a provided array passes a test implemented by a predicate function.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/array/base/cuevery-by/README.md]:
```suggestion
Cumulatively tests whether every array element in a provided array passes a test implemented by a predicate function and assigns the results to a provided output array.
```

File: lib/node_modules/@stdlib/array/base/cuevery-by/README.md [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+var y = [ false, null, false, null, false, null, false, null, false, null ];
+
+var out = cueveryBy.assign( x, y, 2, 0, fcn );
+// returns [ true, null, true, null, false, null, false, null, false, null ]
+
+var bool = ( out === y );
+// returns true
+```
+
+The invoked `predicate` function is provided three arguments:
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/array/base/cuevery-by/README.md:48]: We should move L67-L92 to before the `assign` documentation given it's applicable to both functions.

File: lib/node_modules/@stdlib/array/base/cuevery-by/docs/repl.txt [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+
+    Examples
+    --------
+    > function fcn( v ) { return ( v > 0 ); };
+    > var x = [ 1, 1, 0, 0, 0 ];
+    > var y = {{alias}}( x, fcn )
+    [ true, true, false, false, false ]
+
+
+{{alias}}.assign( x, out, stride, offset, predicate[, thisArg])
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/array/base/cuevery-by/docs/repl.txt]:
```suggestion
{{alias}}.assign( x, out, stride, offset, predicate[, thisArg] )
```

**Review by Planeshifter — CHANGES_REQUESTED** [review]
Thanks for the contribution!

Left a few comments; PR looks good overall.

**Comment by Kaif987** [comment]
Hey @Planeshifter  I just added the changes mentioned.

**Review by Planeshifter — APPROVED** [review]
Thank you, @Kaif987! Will merge shortly after CI has cleared.

**Pull request merged**